functions/src/winloss.ts

```diff
@@ -0,0 +1,125 @@
+import { firestore } from "firebase-admin";
+import * as functions from "firebase-functions";
+import { db } from "./app";
+import { getPrivatePlayerCollection, lobbyCollection } from "./firestore-collections";
+import { isLobbyRequest } from "./firebase-functions-types";
+import { Lobby, PrivatePlayer } from "./firestore-types/lobby";
+
+// this function will likely be obsolete once timers are introduced
+export const lobbyReturn = functions.https.onCall(async (data: unknown, context): Promise<void> => {
+  if (context.auth === undefined) {
+    throw new functions.https.HttpsError("permission-denied", "Not Signed In");
+  }
+
+  // validate code
+  if (!isLobbyRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Invalid lobby code!");
+  }
+
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
+  if (lobby.exists === false) {
+    throw new functions.https.HttpsError("not-found", "Lobby doesn't exist!");
+  }
+
+  const { uids, state } = lobby.data() as Lobby;
+  // check if host
+  if (context.auth.uid !== uids[0]) {
+    throw new functions.https.HttpsError("permission-denied", "Not the host of the game!");
+  }
+
+  // check game state
+  if (state !== "END") {
+    throw new functions.https.HttpsError("invalid-argument", "Wrong game state!");
```

Comment:
`invalid-argument` is fine but in the future `failed-precondition` would probobly be better